### ISSUE TITLE
Implement a `nn.Sequential`-like head layer

### ISSUE THREAD

**Issue by monatis** [qdrant/quaterion#97]
This will allow users to compose custom head layers easily by appending arbitrary `nn.Module`s to it.

For example, they will be able to add a dropout before the actual head layer etc.
